pkg/utils/has/components.go

```diff
@@ -102,7 +102,7 @@ func (h *HasController) WaitForComponentPipelineToBeFinished(component *appservi
 				return true, nil
 			} else {
 				var prLogs string
-				if err = tekton.StorePipelineRun(pr, h.KubeRest(), h.KubeInterface()); err != nil {
+				if err = tekton.StoreFailedPipelineRun(pr, h.KubeRest(), h.KubeInterface()); err != nil {
```

Comment:
I am confused about  the method name `StoreFailedPipelineRun`, this method is to store PipelineRun log, no matter it's failed pipelinerun or not

Reply:
Could you please give me a bit more explanation here? Judging by [this line](https://github.com/redhat-appstudio/e2e-tests/blob/9ce5844bd9baee174a201fc22ed52e13984d21d0/pkg/utils/util.go#L188) the function won't be able to store any logs if the given pipelineRun didn't fail.

Reply:
Based on its parameters, it just handles general PipelineRun object. If you pass `FailedPipelineRunDetails` as parameter,  then `StoreFailedPipelineRun ` make sense

Reply:
The function is only used in a [situation when the pipelineRun fails](https://github.com/redhat-appstudio/e2e-tests/blob/96c6f308e10a32ff927752780046e026535742c5/pkg/utils/has/components.go#L101). I think, we should avoid keeping the name `StorePipelineRun` as this function can't store any given pipelineRun.

Reply:
Extracting `FailedPipelineRunDetails` to pass as a parameter would lead to code duplication if the `StoreFailedPipelineRun` func is used again.

Reply:
cc @psturc 
Could you please give an opinion on this?

Reply:
Ok so the `StoreFailedPipelineRun` function currently saves:
1. PipelineRun object (in yaml format)
1. Logs only from failed TaskRun pod (**not all logs** for a given PipelineRun, as one might think that this is the case based on the function name and its description)

If we'd pass the successful PipelineRun as a parameter into this function, it'd still store the PipelineRun object, but not any of the logs, so naming this function `StorePipelineRun` would be also confusing.

My suggestion is to update the [`StoreFailedPipelineRun` function](https://github.com/redhat-appstudio/e2e-tests/pull/688/files#diff-dda2163aa8e3e9cb6cd6f38a5e3df0a184e263ff34e8b7ee17ebe635d83325e1R279) in a way that it extracts all logs from the PipelineRun (i.e. all TaskRun pods related to that PipelineRun) and then rename that function to `StorePipelineRun`.

So in the end `StorePipelineRun` function will store the PipelineRun object in yaml format and it will store **all related logs**.

Then we could also use this updated `StorePipelineRun` function to be called from [`StorePipelineRuns` function](https://github.com/redhat-appstudio/e2e-tests/pull/688/files#diff-dda2163aa8e3e9cb6cd6f38a5e3df0a184e263ff34e8b7ee17ebe635d83325e1R324), which will store all pipelineruns objects and logs in a given namespace. Wdyt?

Reply:
Thanks! I will go ahead with this.

Reply:
[Implemented](https://github.com/redhat-appstudio/e2e-tests/pull/688#discussion_r1302784279).